shoop/core/order_creator/_source.py

```diff
@@ -392,6 +395,19 @@ def get_product_lines(self):
         product_lines = [l for l in self.get_lines() if l.product]
         return product_lines
 
+    def get_suppliers(self):
+        return set([l.supplier for l in self.get_lines() if l.supplier])
```

Comment:
Using `get_lines` here doesn't account for product lines added by OrderSourceModifiers. They could adds some new product lines in the `get_final_lines` phase. (E.g. if there's a campaign that you get a rubber duck for free when buying two laptops, then `get_final_lines` could add the rubber duck to basket if there is two laptops in there.)

Therefore this and the `get_products_and_quantities` should use `get_final_lines` rather than `get_lines`, but does it create a cycle somewhere?